aio/aio-proxy/aio_proxy/search/execute_search.py

```diff
@@ -62,5 +48,71 @@ def sort_and_execute_search(
             matching_unite_legale_dict["matching_etablissements"] = []
 
         response.append(matching_unite_legale_dict)
+    search_response = {
+        "total_results": total_results,
+        "response": response,
+        "execution_time": execution_time,
+        "include_etablissements": include_etablissements,
+    }
+    return search_response
+
+
+def sort_search(search, is_text_search: bool):
+    # Sorting is very heavy on performance if there is no
+    # search terms (only filters). As there is no search terms, we can
+    # exclude this sorting because score is the same for all results
+    # documents. Beware, nom and prenoms are search fields.
+    if is_text_search:
+        search = search.sort(
+            {"_score": {"order": "desc"}},
+            {"etat_administratif_unite_legale": {"order": "asc"}},
+        )
+    # If only filters are used, use nombre établissements ouverts to sort the results
+    else:
+        search = search.sort(
+            {"nombre_etablissements_ouverts": {"order": "desc"}},
+        )
+    return search
+
+
+def sort_and_execute_search(
+    search,
+    offset: int,
+    page_size: int,
+    is_text_search: bool,
+    include_etablissements: bool,
+) -> dict:
+    search = search.extra(track_scores=True)
+    search = search.extra(explain=True)
+    # Collapse is used to aggregate the results by siren. It is the consequence of
+    # separating large documents into smaller ones
+    search = search.update_from_dict({"collapse": {"field": "siren"}})
+    search = sort_search(search, is_text_search)
+
+    # Execute search
+    def get_search_response():
```

Comment:
`should_cache_search_response` does not need to be an inner function since its execution is entirely independent of the `sort_and_execute_search` function logic.
It does however depend on the `search.execute()`  output, because it uses the execution time to return a value.
`get_search_response()` is a call function for `execute_and_format_search_response` to avoid having to give the function parameters when calling `execute_and_format_search_response` in `cache_strategy` (i.e `include_etablissements, `offset`etc ), which is not the case for `should_cache_search_response` that does not need a predefined parameter and depends on the search execution.